Document Elasticsearch member search (#7068)
hammerandchisel/discord-api-docs: docs/resources/Guild.md

> info
> BYPASSES_VERIFICATION allows a member who does not meet verification requirements to participate in a server.

### Supplemental Guild Member Object

> Additional information about a participating user's join source in a [guild](#DOCS_RESOURCES_GUILD/guild-object).

Reviewer: ```suggestion
Additional information about a user's join source in a [guild](#DOCS_RESOURCES_GUILD/guild-object).
```

| member             | [guild member](#DOCS_RESOURCES_GUILD/guild-member-object) object | The associated guild member                                                                              |
| join_source_type   | integer                                                          | [How the user joined the guild](#DOCS_RESOURCES_GUILD/supplemental-guild-member-object-join-source-type) |
| source_invite_code | ?string                                                          | The invite code or vanity used to join the guild, if applicable                                          |
| inviter_id         | ?snowflake                                                       | The ID of the user who invited the user to the guild, if applicable                                      |
| integration_type?  | ?integer                                                         | The type of integration that added the user to the guild, if applicable                                  |

###### Join Source Type

> | Value | Name                       | Description                                                                                                             |

Reviewer: Same as below, the first two columns should probably be switched...

| query | string  | Query string to match username(s) and nickname(s) against. |         |
| limit | integer | max number of members to return (1-1000)                   | 1       |

# Search Guild Members % POST /guilds/{guild.id#DOCS_RESOURCES_GUILD/guild-object}/members-search

Returns [supplemental guild member](#DOCS_RESOURCES_GUILD/supplemental-guild-member-object) objects containing [guild member](#DOCS_RESOURCES_GUILD/guild-member-object) objects that match a specified query. Requires the `MANAGE_GUILD` permission.

This endpoint utilizes Elasticsearch to power results. This means that while it is very powerful, it's also tricky to use and reliant on the index, meaning results may not be immediately available for a recently-joined member.

> > warn

Reviewer: Do code blocks work inside warn boxes?

Author: hope so

| or_query?  | [member filter](#DOCS_RESOURCES_GUILD/search-guild-members-member-filter-structure) object                       | The filter criteria to match against members using OR logic                                                            |
| and_query? | [member filter](#DOCS_RESOURCES_GUILD/search-guild-members-member-filter-structure) object                       | The filter criteria to match against members using AND logic                                                           |
| before?    | [member pagination filter](#DOCS_RESOURCES_GUILD/search-guild-members-member-pagination-filter-structure) object | Get members before this member                                                                                         |
| after?     | [member pagination filter](#DOCS_RESOURCES_GUILD/search-guild-members-member-pagination-filter-structure) object | Get members after this member                                                                                          |

###### Member Sort Type

> | Value | Name           | Description                                                 |

Reviewer: I think constants are usually documented using Name - Value. Also maybe name could be replaced with sort type

| user_id?            | [query](#DOCS_RESOURCES_GUILD/search-guild-members-query-structure) object (snowflake) | Query to match member IDs against                                                                                                        | `or_query`, `range`     |
| usernames?          | [query](#DOCS_RESOURCES_GUILD/search-guild-members-query-structure) object (string)    | Query to match display name(s), username(s), and nickname(s) against                                                                     | `or_query`              |
| role_ids?           | [query](#DOCS_RESOURCES_GUILD/search-guild-members-query-structure) object (snowflake) | IDs of roles to match members against                                                                                                    | `or_query`, `and_query` |
| guild_joined_at?    | [query](#DOCS_RESOURCES_GUILD/search-guild-members-query-structure) object (integer)   | When the user joined the guild                                                                                                           | `range`                 |

> | safety_signals?     | [safety signals](#safety-signals-structure) object                                     | Safety signals to match members against                                                                                                  |

Reviewer: Isn't this row missing the last column?

Author: formatter stripped it because the value is empty

| unusual_account_activity?     | boolean                                                                              | Whether unusual account activity is detected                                                                                                                                         | `true`, `false` |
| automod_quarantined_username? | boolean                                                                              | Whether the member has been indefinitely quarantined by [an AutoMod Rule](#DOCS_RESOURCES_AUTO_MODERATION/auto-moderation-rule-object) for their username, display name, or nickname | `true`, `false` |

###### Query Structure

| Field      | Type                                                                                   | Description                                                            |
|------------|----------------------------------------------------------------------------------------|------------------------------------------------------------------------|

> | or_query?  | array[snowflake \| string \| integer]                                                  | The values to match against using OR logic (1-100 characters, max 10)  |

Reviewer: ```suggestion
| or_query?  | array[snowflake \| string \| integer]                                                  | The values (up to 10) to match against using OR logic (1-100 characters each)  |
```
I think "array of" (instead of `array[]`) and "or" (instead of `|`) would be more in line with the usual docs style...

> | Field      | Type                                                                                   | Description                                                            |
> |------------|----------------------------------------------------------------------------------------|------------------------------------------------------------------------|
> | or_query?  | array of snowflakes,  strings, or integers                                             | The values to match against using OR logic (1-100 characters, max 10)  |
> | and_query? | array of snowflakes,  strings, or integers                                             | The values to match against using AND logic (1-100 characters, max 10) |
> | range?     | [range query](#DOCS_RESOURCES_GUILD/search-guild-members-range-query-structure) object | The range of values to match against                                   |

Reviewer: ```suggestion
| Field      | Type                                                                                   | Description                                                            |
|------------|----------------------------------------------------------------------------------------|------------------------------------------------------------------------|
| or_query?  | array of snowflakes, strings, or integers                                              | The values to match against using OR logic (1-100 characters, max 10)  |
| and_query? | array of snowflakes, strings, or integers                                              | The values to match against using AND logic (1-100 characters, max 10) |
| range?     | [range query](#DOCS_RESOURCES_GUILD/search-guild-members-range-query-structure) object | The range of values to match against                                   |
```

###### Range Query Structure

> | Field | Type                 | Description                          |
> |-------|----------------------|--------------------------------------|
> | gte?  | snowflake \| integer | Inclusive lower bound value to match |
> | lte?  | snowflake \| integer | Inclusive upper bound value to match |

Reviewer: ```suggestion
| Field | Type                 | Description                          |
|-------|----------------------|--------------------------------------|
| gte?  | snowflake or integer | Inclusive lower bound value to match |
| lte?  | snowflake or integer | Inclusive upper bound value to match |
```
